### ISSUE TITLE
Possible to initialize an Automerge.Table from existing data?

### ISSUE THREAD

**Issue by SahidMiller** [automerge/automerge#275]
Initializing an Automerge.Table from existing relational data (serialized from Automerge document) would be very helpful for sharing relational data across documents without complex migration instructions to maintain relationships. 

My use case is shipping content created in my own Automerge instance/network as a package for otherwise unrelated Automerge networks/instances. Keeping another set of unique keys for maintaining relationships between entities is an option but takes away a lot of the convenience of Automerge.Table.

**Comment by ept** [comment]
Hi @SahidMiller! You can repeatedly call `.add(...)` on the table within a `change` callback in order to populate it with your existing data. Or does that not do what you need?

**Comment by SahidMiller** [comment]
Hello, @ept! No, unfortunately not. I'm using a CMS-like application that relies heavily on Automerge.Table to build a number of related entities referencing each other by their ids. I'd like to serialize some of the entities I have on my instance so they can be packaged and published for others to install on their sites. The installer would not be collaborating with the package creator in any meaningful sense so reusing the same ids seems justified, especially considering the work to orchestrate if id's did change and the references would need to be updated. (since it's cms-like, it's not easy to get all the references upfront in some cases)

Although I used the word "initialize" in the title and was expecting to use something similar to `Automerge.from`, it also applies for cases where a package might be installed after an Automerge document is already initialized. It was just that my use case involves building "starter kits" for new users to install when starting their CMS.

**Comment by ept** [comment]
Maybe this will work for you: rather than initialising a fresh Automerge document on each node, can you create the table and its contents once, use `Automerge.save()` to write out the document, and then distribute that serialised document to everyone? Each node that loads the document will then see the table with the same contents, and with the same ID for each element.

**Comment by SahidMiller** [comment]
Automerge.save would snapshot the entire history which is useful for users in the same CMS room (which I was calling a network/instance originally). Whereas I was hoping for clients to choose a subset of their content to publish as a package for other CMS rooms to install, either at initialization or after initialization.

**Comment by ept** [comment]
Sorry, I still don't quite follow what you are trying to do. Do you want for clients to sync only some of the rows in a table? If so, how do you determine which ones should be synced? Please could you give a bit more detail.

You might be able to construct this by using a separate actorId for each subset to be synced, and then when you do `Automerge.getAllChanges()` you can select just those you want. However, changes can only be applied if their dependencies are satisfied, so you would have to construct the changes in a way such that one actorId does not depend on another. This is possible, but requires some care.

**Comment by SahidMiller** [comment]
That's interesting, I haven't heard of choosing a subset of commits, thanks for the suggestion! I don't actually need the history when exporting those rows, I simply want to maintain their primary keys (the table row id) so the relationships are maintained. 

Sorry if I wasn't clear enough though, I figured it would be a bit confusing. A good analogy might be copy-pasting from one collaborative document to another unrelated collaborative document, we don't want the history of that slice of data to be copied over or to have any collaborative connection between the two documents, just preserve WHAT we copied over so it works in the application we paste it. In my case, it's complex entities which have relationships via row ID.
